pkg/controller/master/virtualmachine/vm_controller.go

```diff
@@ -398,6 +404,48 @@ func (h *VMController) SetHaltIfInsufficientResourceQuota(_ string, vm *kubevirt
 	return vm, h.stopVM(vm, err.Error())
 }
 
+// RestartFromPausedIOError automatically restarts VMs from status flipping between Paused and Running
+// As a workaround for the issue https://github.com/harvester/harvester/issues/4633
+func (h *VMController) RestartFromPausedIOError(_ string, vm *kubevirtv1.VirtualMachine) (*kubevirtv1.VirtualMachine, error) {
+	if vm == nil || vm.DeletionTimestamp != nil || vm.Spec.Template == nil {
+		return vm, nil
+	}
+
+	// if the vm is not paused or it's under restarting, skip it.
+	if vm.Status.PrintableStatus != kubevirtv1.VirtualMachineStatusPaused || len(vm.Status.StateChangeRequests) != 0 {
+		return vm, nil
+	}
+
+	vmi, err := h.vmiCache.Get(vm.Namespace, vm.Name)
```

Comment:
skip the vmi none-existing error ?

Reply:
Yes, we check whether VM is paused at first. In a happy path, a paused VM has related VMI CR. If we can't find VMI in this case, there must have some errors.